src/tools/cephfs_mirror/PeerReplayer.cc

```diff
@@ -1481,19 +1475,17 @@ int PeerReplayer::do_synchronize(const std::string &dir_root, const Snapshot &cu
       if ("." == nname || ".." == nname)
         continue;
       // create path for the newly found entry
-      npath = entry_diff_path(epath, nname);
-      nabs_path = entry_diff_path(dir_root, npath);
-
-      r = ceph_statx(sd_info.cmount, nabs_path.c_str(), &cstx,
-                     CEPH_STATX_MODE | CEPH_STATX_UID | CEPH_STATX_GID |
-                     CEPH_STATX_SIZE | CEPH_STATX_ATIME | CEPH_STATX_MTIME,
-                     AT_STATX_DONT_SYNC | AT_SYMLINK_NOFOLLOW);
+      npath = entry_path(epath, nname);
```

Comment:
@joscollin Could you share a sample log snippet involving these paths without and with the fix?

Reply:
@joscollin - if possible please share this, otherwise I will check it when I pull in this change to test the file block diff integration with the mirror daemon.

Reply:
@vshankar Let me see, if I could reproduce it. I don't have a build now.  So please don't wait for me. If you are testing block diff, please pull this PR.